Source/missiles.cpp

```diff
@@ -1393,11 +1393,10 @@ void AddStealMana(Missile &missile, AddMissileParameter & /*parameter*/)
 
 void AddSpectralArrow(Missile &missile, AddMissileParameter &parameter)
 {
+	Player &player = *missile.sourcePlayer();

```

Comment:
This dereferences a null pointer if `missile.sourceType() == MissileSource::Player`.
This line should be moved back to where it was before.